src/Maestro/DependencyUpdateErrorProcessor/DependencyUpdateErrorProcessor.cs

```diff
@@ -0,0 +1,227 @@
+            {
+                try
+                {
+                    if (!string.IsNullOrEmpty(whereToCreateIssue))
+                    {
+                        IGitHubTokenProvider gitHubTokenProvider = Context.GetService<IGitHubTokenProvider>();
+                        long installationId = await Context.GetInstallationId(whereToCreateIssue);
+                        gitHubToken = await gitHubTokenProvider.GetTokenForInstallationAsync(installationId);
+
+                        Logger.LogInformation($"GitHub token acquired for '{whereToCreateIssue}'!");
+                    }
+
+                    IssueManager issueManager = new IssueManager(gitHubToken, azureDevOpsToken);
+
+                    string title = $"Repository Url :'{repositoryUrl}' Error :'{errorMessage}'";
+                    string description = $"Something failed during dependency update" +
+                    $"{Environment.NewLine} {Environment.NewLine}" +
+                    "RepositoryUrl :" + $"[{repositoryUrl}](repositoryUrl)" +
+                    $"{Environment.NewLine} {Environment.NewLine}" +
+                    "Branch Name :" + $"{branch}" +
+                    $"{Environment.NewLine} {Environment.NewLine}" +
+                    "Error Message : " + $"{errorMessage}" +
+                    $"{Environment.NewLine} {Environment.NewLine}" +
+                    "Method : " + $"{method}" +
+                    $"{Environment.NewLine} {Environment.NewLine}" +
+                    "Action : " + $"{action}" +
+                    $"{Environment.NewLine} {Environment.NewLine}" +
+                    "Failure Category :" + $"{label}" + // Work on failure category will be done as a part of -> https://github.com/dotnet/arcade/issues/1196
+                    $"{Environment.NewLine} {Environment.NewLine}" +
+                    $"/FYI: {fyiHandles}";
+
+
+                    Logger.LogInformation($"GitHub token acquired for '{whereToCreateIssue}'!");
+                    //Check if the issue already exists in Github
+                    if (await SearchCreatedIssueAsync(transactionDateTime, errorMessage))
+                    {
+                        int issueId = await issueManager.CreateNewIssueAsync(whereToCreateIssue, title, description);
+                        Logger.LogInformation($"Issue {issueId} was created in '{whereToCreateIssue}'");
+                        Logger.LogInformation("have to create issue");
+                    }
+                    else
+                    {
+                        Logger.LogInformation("Issue already exists for this particular error");
+                    }
+                }
+                catch (Exception exc)
+                {
+                    Logger.LogError(exc, $"Something failed while attempting to create an issue based on repo '{repositoryUrl}' ");
+                }
+            }
+        }
+
+        public Task<TimeSpan> RunAsync(CancellationToken cancellationToken)
```

Comment:
Is... throwing from this method ok?

Reply:
A comment explaining what it means for this method to throw might make sense, but it seems like we've got a bad abstraction here... we are implementing an interface... but literally the only method in it we aren't implementing.